helm-chart/kuberay-operator/templates/deployment.yaml

```diff
@@ -42,10 +42,10 @@ spec:
             {{- $argList = append $argList "--enable-batch-scheduler" -}}
             {{- end -}}
             {{- $watchNamespace := "" -}}
-            {{- if .Values.singleNamespaceInstall -}}
+            {{- if and .Values.singleNamespaceInstall (not .Values.watchNamespace) -}}
             {{- $watchNamespace = .Release.Namespace -}}
```

Comment:
Case 1: If `singleNamespaceInstall` is set to true and `watchNamespace` is not specified, the KubeRay operator will only watch the namespace where the operator is deployed.

Case 2: If `watchNamespace` is specified, the KubeRay operator will watch all namespaces listed in the variable. In this case, we concatenate all namespaces into a string separated by commas. Then, the string will be parsed by `strings.Split(watchNamespace, ",")` in main.go.

Reply:
> But I think the old behavior of watchNamespace is missing from the PR description, do you mind adding it?

The behavior is described by the comments in `values.yaml`. The comments are deleted by this PR.

```yaml
# kuberay operator will only watch the resource events from the "watchNamespace" namespace.
# this option has no effect if singleNamespaceInstall is true, because we assume there are no
# permissions outside of the current namespace
# watchNamespace: ray-user-namespace
```

> Just to double check, are commas , forbidden characters in namespaces?

The flag `--watch-namespace` is a list of namespaces, separated by commas. It will be parsed by `watchNamespaces := strings.Split(watchNamespace, ",")` in `main.go`.

Reply:
> Right, but if a namespace is allowed to have a comma in it then breaking by commas won't work.

Oh, I understand now. I missed that part. Thank you for pointing it out!